R/opts.R

```diff
@@ -638,9 +638,23 @@ obs_opts <- function(family = c("negbin", "poisson"),
       .frequency = "regularly",
       .frequency_id = "obs_opts"
     )
-  #nolint end
+  } else {
```

Comment:
```suggestion
  } else if (missing(na)) {
```

I think we can assume that if a user sets this they know what they're doing. I also think that we should check here if there are any missing dates or NA cases as otherwise we're just flooding the user with unneeded information.

Reply:
This turned out to be more complicated than I thought.

I've made the following changes:
- Added a testing function, `test_data_complete()` to check whether the data is complete or incomplete (has explicit or implicit missingness). See f36ec5a8132fd1d88dd51e64d4a3c50142186119. This function was needed anyway and can be used in conjunction with `create_complete_cases()` so that the latter is not called if not needed.
- Added a new return variable to `obs_opts()` which checks if the user specified `na = "missing"` for use with the next function, `cross_check_na_against_data()`.
- Why `cross_check_na_against_data()`?  `obs_opts()` does not injest the data, so it has no way of knowing its nature (complete or not), so `cross_check_na_against_data()` throws a message if the data is incomplete and the default `na = "missing"` is being used. Any other combination does not throw a message.
- These changes mean that the message is thrown from the main function that calls `obs_opts()`. The alternative would have been for `obs_opts()` to gain a new argument that has information on whether the data is complete or not but that makes no sense in the stand-alone `obs_opts()` function.
- I've added tests for all these functions.